pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertyDescriptorFactory.java

```diff
@@ -28,30 +30,42 @@
  */
 public abstract class AbstractPropertyDescriptorFactory<T> implements PropertyDescriptorFactory<T> {
 
-    protected static final Map<PropertyDescriptorField, Boolean> CORE_FIELD_TYPES_BY_KEY
-        = CollectionUtil.mapFrom(new PropertyDescriptorField[] {NAME, DESCRIPTION, DEFAULT_VALUE, DELIMITER},
-                                 new Boolean[] {true, true, true, false});
+
+
+
+    static {
+        System.err.println(CORE_FIELD_TYPES_BY_KEY);
```

Comment:
I thought it was a cycle too, but I after some verification, IntegerProperty has no dependency on PropertyDescriptorUtil. The class is entirely similar to eg `FloatProperty`, but it's initialized after PropertyDescriptorUtil accesses its factory.
To illustrate, I added the following snippet right after the `FACTORY` declaration of FloatProperty and IntegerProperty:
```java
static {
  System.out.println("FACTORY INITIALISED: " + FACTORY);
}
``` 
and I added logging statements to the PropertyDescriptor initialiser:
```java
static {
  ...
  temp.put("Integer", IntegerProperty.FACTORY);
  System.out.println("DONE WITH INTEGER FACTORY:" + IntegerProperty.FACTORY);
  ...
  temp.put("Float", FloatProperty.FACTORY);
  System.out.println("DONE WITH FLOAT FACTORY:" + FloatProperty.FACTORY);
}
```
Console output starts with this:
```
DONE WITH INTEGER FACTORY:null
FACTORY INITIALISED: net.sourceforge.pmd.lang.rule.properties.IntegerProperty$1@123a439b
FACTORY INITIALISED: net.sourceforge.pmd.lang.rule.properties.FloatProperty$1@379619aa
DONE WITH FLOAT FACTORY: net.sourceforge.pmd.lang.rule.properties.FloatProperty$1@379619aa
```
The integer factory is initialized after having been accessed, which makes no sense (see [this SO answer](https://stackoverflow.com/a/2007701/6245827)). I have no idea where the bug comes from, but a workaround would be to use a static method returning the factory, with the downside that this factory would not be a singleton (though this method could be made package private). It looks sloppy though :/